library/test/src/lib.rs

```diff
@@ -514,6 +516,21 @@ pub fn filter_tests(opts: &TestOpts, tests: Vec<TestDescAndFn>) -> Vec<TestDescA
     filtered
 }
 
+/// Custom handling of cases where there is only 1 test to be executed.
+/// This is needed because in such cases we may want to show more detailed information.
+/// For example for investigation purposes we may want to show the reasons why the test case was ignored.
+fn check_for_single_test_run(filtered_tests: &Vec<TestDescAndFn>) {
```

Comment:
These changes to libtest also seem useful, but could you move them to a separate PR and only change compiletest here?